trace/doc.go

```diff
@@ -0,0 +1,82 @@
+// Package trace gives access to direct control over the span and trace objects
+// used by the beeline.
+//
+// Summary
+//
+// Elementary use of the Honeycomb beeline has few needs - use a wrapper to
+// cover most of the basic info about a given event and then augment that event
+// with a few fields using `beeline.AddField`. Occasionally add an additional
+// span to the trace that is automatically created and add fields to that.
+// However, as the beeline starts being used in more sophisticated applications,
+// or new wrapers are being written, more direct control is needed to manipulate
+// the spans and traces generated by the beeline.  Using the `trace` package
+// enables this more sophisticated use.
+//
+// The types of use that are enabled by using the trace package:
+//
+// 1) creating asynchronous spans; spans that outlive the main trace. These spans
+// are especially useful for goroutines that manage some background task after
+// the main user-facing chore has been completed. Examples of this are sending
+// an email or persisting accepted data.
+//
+// 2) creating traces with upstream parents when you are a downstream service.
+// The existing HTTP wrappers do this for you, but if your trace is getting
+// propagated via kafka or SQS or some other mechanism you may need to do this
+// yourself
+//
+// 3) adding fields that use a different naming scheme. Fields added via the
```

Comment:
this feels like a bad idea when we depend on column names and their values for our server-side data model.

I'm not saying we'll see collisions immediately but we'll have to start actually mining customer dataset schemas to see if we will whenever we add a new beeline-specific column.

Reply:
Part of the separation between the beeline / trace packages is that using the trace package gives you more flexibility (and therefore more possibility you break things). This part of the API is necessary for the wrappers (that create fields with `http.*` and `db.*`) to work without being extra privileged.  In order to make it possible for other people to write their own wrappers that are as effective as ours, this ability must be exposed somehow. This feels like the right place to put it.

Reply:
Sure, there needs to be two paths, but we can have an api that users use and an api that instrumentations use.  I don't think we should have a single api that both use that permits collisions like this.